pkg/utils/utils.go

```diff
@@ -23,3 +23,26 @@ func GetMapValue(key string, dataMap map[string][]byte) (string, bool) {
 	}
 	return "", false
 }
+
+// Determine whether the label is attached to the runner or the reporter.
+func AddMapValue(key string, value string, dataMap map[string]string, overwrite bool) map[string]string {
+	_, ok := dataMap[key]
+	if overwrite {
+		if ok {
+			dataMap[key] = value
+			return dataMap
+		} else {
+			dataMap = map[string]string{}
+			dataMap[key] = value
+			return dataMap
+		}
+	} else {
+		if ok {
+			return dataMap
+		} else {
+			dataMap = map[string]string{}
+			dataMap[key] = value
+			return dataMap
+		}
+	}
```

Comment:
@S-mishina  there are 2 things:

- 1. have you tested the code? I don't get an expected result with your code (please see the test code below) 
- 2. it's very redundant ( you can make it alot slimmer )

# test code

```go
package main

import "fmt"

func AddMapValue(key string, value string, dataMap map[string]string, overwrite bool) map[string]string {
	_, ok := dataMap[key]
	if overwrite {
		if ok {
			dataMap[key] = value
			return dataMap
		} else {
			dataMap = map[string]string{}
			dataMap[key] = value
			return dataMap
		}
	} else {
		if ok {
			return dataMap
		} else {
			dataMap = map[string]string{}
			dataMap[key] = value
			return dataMap
		}
	}
}

func main() {
	m := map[string]string{"apple": "applevalue", "banana": "bananavalue", "lemon": "lemonvalue"}
	fmt.Printf("%v\n", AddMapValue1("pine", "pinevalue", m, false)) 
        // actual result ->   map[pine:pinevalue]
        // expected result ->   map[apple:applevalue banana:bananavalue lemon:lemonvalue pine:pinevalue]

}

```

Reply:
Here is my suggestion:

```go
func AddMapValue(key string, value string, dataMap map[string]string, overwrite bool) map[string]string {
	if _, ok := dataMap[key]; ok && !overwrite {
		return dataMap
	}
	dataMap[key] = value
	return dataMap
}
```

# testing code

```
package main

import "fmt"

func AddMapValue(key string, value string, dataMap map[string]string, overwrite bool) map[string]string {
	if _, ok := dataMap[key]; ok && !overwrite {
		return dataMap
	}
	dataMap[key] = value
	return dataMap
}

func main() {
	m := map[string]string{"apple": "applevalue", "banana": "bananavalue", "lemon": "lemonvalue"}
	fmt.Printf("%v\n", AddMapValue("pine", "pinevalue", m, false))
	fmt.Printf("%v\n", AddMapValue("lemon", "pinevalue", m, false))
	fmt.Printf("%v\n", AddMapValue("lemon", "pinevalue", m, true))
}
```
output was ok as expected

```
map[apple:applevalue banana:bananavalue lemon:lemonvalue pine:pinevalue]
map[apple:applevalue banana:bananavalue lemon:lemonvalue pine:pinevalue]
map[apple:applevalue banana:bananavalue lemon:pinevalue pine:pinevalue]
```